src/openzeppelin/tests/test_account.cairo

```diff
@@ -0,0 +1,330 @@
+
+    let supports_account_interface: bool = Account::supports_interface(ERC165_ACCOUNT_ID);
+    assert(supports_account_interface, 'Should support account id');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_is_valid_signature() {
+    let data = SIGNED_TX_DATA();
+    let message = data.transaction_hash;
+
+    let mut good_signature = ArrayTrait::new();
+    good_signature.append(data.r);
+    good_signature.append(data.s);
+
+    let mut bad_signature = ArrayTrait::new();
+    bad_signature.append(0x987);
+    bad_signature.append(0x564);
+
+    Account::set_public_key(data.public_key);
+
+    let is_valid = Account::is_valid_signature(message, good_signature);
+    assert(is_valid == ERC1271_VALIDATED, 'Should accept valid signature');
+
+    let is_valid = Account::is_valid_signature(message, bad_signature);
+    assert(is_valid == 0_u32, 'Should reject invalid signature');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_validate_deploy() {
+    let class_hash = 0x123;
+    let salt = 0;
+    let public_key = 0x234;
+    let account = setup_dispatcher(Option::Some(@SIGNED_TX_DATA()));
+
+    assert(
+        account.__validate_deploy__(class_hash, salt, public_key) == starknet::VALIDATED,
+        'Should validate correctly'
+    );
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_validate_declare() {
+    let class_hash = 0x123;
+    let account = setup_dispatcher(Option::Some(@SIGNED_TX_DATA()));
+
+    assert(
+        account.__validate_declare__(class_hash) == starknet::VALIDATED, 'Should validate correctly'
+    );
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_execute() {
+    let data = SIGNED_TX_DATA();
+    let account = setup_dispatcher(Option::Some(@data));
+    let initial_public_key = data.public_key;
+    let mut calls = ArrayTrait::new();
+
+    let set_public_key_selector = 0x2e3e21ff5952b2531241e37999d9c4c8b3034cccc89a202a6bf019bdf5294f9;
+
+    assert(account.get_public_key() == initial_public_key, 'Should get initial public key');
+
+    let mut calldata = ArrayTrait::new();
+    let new_public_key = 0x789789;
+    calldata.append(new_public_key);
+    let call = Call {
+        to: account.contract_address, selector: set_public_key_selector, calldata: calldata
+    };
+
+    calls.append(call);
+    let ret = account.__execute__(calls);
+
+    assert(account.get_public_key() == 0x789789, 'Should get new public key');
+
+    // Test return value
+    let mut call_retval = ret.at(0).span();
+    assert(call_retval.len() == 0, 'Should be an empty response');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_validate() {
+    let calls = ArrayTrait::new();
+    let account = setup_dispatcher(Option::Some(@SIGNED_TX_DATA()));
+
+    assert(account.__validate__(calls) == starknet::VALIDATED, 'Should validate correctly');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_multicall() {
+    let account = setup_dispatcher(Option::Some(@SIGNED_TX_DATA()));
+    let erc20 = deploy_erc20(account.contract_address, 1000);
```

Comment:
if we decide to drop the account deployment, we can simply use a const address for minting, then we can set it as tx sender

Reply:
Following on [this comment](https://github.com/OpenZeppelin/cairo-contracts/pull/616#discussion_r1192472762), personally, I wouldn't say I like the idea of removing the deployment of the account, to me is getting further away from the actual flow, and in some cases, we can't.

Reply:
`execute` as stated in [this comment](https://github.com/OpenZeppelin/cairo-contracts/pull/616#discussion_r1196468098).

Reply:
IMO the complexity added on top is pretty small, and at least having the entry point executed by a syscall using the setup_dispatcher (setting the contract_address and the caller_address automatically) is less error-prone (even when is still quite far from the real flow, is closer). But there is a line ofc, we can use direct calls (instead of the dispatcher) for simple logic like Account::_is_valid_signature, which is not an entry point or context, but we need to use dispatcher to test interactions like execute, I cannot say exactly what we should use in the middle, but for me looks better using dispatcher for entry points like validate, even when we could use direct calls